internal/cache/in_memory.go

```diff
@@ -0,0 +1,121 @@
+package cache
+
+import (
+	"errors"
+	"sync"
+	"time"
+)
+
+type MemoryStore struct {
+	data sync.Map
+}
+
+func NewMemoryStore() *MemoryStore {
+	return &MemoryStore{}
+}
+
+// Get retrieves a value by key and hash (if given)
+func (s *MemoryStore) Get(hash *string, key string) ([]byte, error) {
+	if hash != nil {
```

Comment:
This `hash` is optional, as I understand, but because of that, in every method we have to check `if hash != nil`. Also, currently, we're always using a hash and never access the general store.

What do you think about defining a list of `allowedHashSets` (e.g. in `cache.go`) with hashes of sets that we can use and only allow to use those HSets, instead of allowing using the general store? Or even better - just define those allowable sets in an Enum, which we can use everywhere in code.

Then, you could also check if given hash exists in `allowedHashSets` (or enum) and if not - return an ErrNotAllowedSet or something like that. You wouldn't have to retrieve values in a different way depending on whether the hash exists or not and it would not have to be a pointer.

Reply:
Example with enum:
```go
type HashSet string

const (
  CacheStatusUpdateHashSet HashSet = "status-update"
  OtherServiceHashSet      HashSet = "different-hash-set"
)

...

func (c *Cache) Get(hash HashSet, key string) (value []byte, err error) {
  ...
}
```